### PR TITLE
ACS-7604: Switch to our custom ClientCredentialsInterceptor 

### PR DESCRIPTION
**Pull request by robinmohlePOP**
ACS-7604: Switch to our custom ClientCredentialsInterceptor that will retrieve a token if needed

### CODE REVIEW

File: src/main/java/org/miracum/etl/fhirtoomop/interceptors/ClientCredentialsInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    this.clientSecret = secret;
+    this.tokenUrl = tokenUrlIn;
+    this.tokenMinValidityBuffer = tokenMinValidityBuffer;
+  }
+
+  @Override
+  public void interceptRequest(IHttpRequest theRequest) {
+    log.info("ClientCredentialsIntercepter has intercepted the request");
+    log.info("ClientId: " + clientId);
+    log.info("clientSecret: " + clientSecret);
```
Comment (michealharrington) [inline src/main/java/org/miracum/etl/fhirtoomop/interceptors/ClientCredentialsInterceptor.java]: Should this be `debug` level? We don't want all these logs every request

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -32,6 +32,8 @@ data:
     baseUrl: ""
     username: ""
     password: ""
+    tokenUrl: ""
+    tokenMinValiditySeconds: 20
```
Comment (michealharrington) [inline src/main/resources/application.yml:36]: Rather than sprinkling the default of `20` everywhere, why not default it in the `FhirConfig.java`?

**Review by michealharrington — CHANGES_REQUESTED** [review]

File: src/main/java/org/miracum/etl/fhirtoomop/interceptors/ClientCredentialsInterceptor.java (hunk 1 above)
Comment (robinmohlePOP) [reply src/main/java/org/miracum/etl/fhirtoomop/interceptors/ClientCredentialsInterceptor.java]: I had them to help me debug, don't need them now.

File: src/main/resources/application.yml (hunk 2 above)
Comment (robinmohlePOP) [reply src/main/resources/application.yml:36]: I may not understand how the application.yml works then... I thought this was where defaults were set?

Comment (fingoltin) [reply src/main/resources/application.yml:36]: I think it would be weird to have a property missing in the yamls.

**Review by fingoltin — APPROVED** [review]

**Review by michealharrington — APPROVED** [review]

**Comment by michealharrington** [comment]
Instead of writing all the code to manage "tokens", is there any way we can enable basic auth via client id + secret?
https://www.keycloak.org/docs/latest/securing_apps/index.html#_java_adapter_config:~:text=is%20false.-,enable%2Dbasic%2Dauth,-OPTIONAL.%20This%20tells

https://stackoverflow.com/a/57819293/1765975

I am mostly worried that RAWA will **not** want to implement similar token logic.
Happy to discuss.

**Comment by robinmohlePOP** [comment]
> I am mostly worried that RAWA will **not** want to implement similar token logic.

I consider Basic Auth to be very insecure. That's not a discussion with me, that's a discussion with Nick/the investors.

**Pull request merged**